### ISSUE TITLE
[Bug]: MultiModalLLMCompletionProgram (Ollama:Llava) raises AttributeError: 'GenerateResponse' object has no attribute 'items'

### ISSUE THREAD

**Issue by baxteran** [run-llama/llama_index#17105]
### Bug Description

MultiModalLLMCompletionProgram  raises AttributeError: 'GenerateResponse' object has no attribute 'items' on inference.

Env: 

pip list| grep -i llama
llama-cloud                              0.1.5
llama-index                              0.12.2
llama-index-agent-openai                 0.4.0
llama-index-cli                          0.4.0
llama-index-core                         0.12.2
llama-index-embeddings-fastembed         0.3.0
llama-index-embeddings-openai            0.3.1
llama-index-indices-managed-llama-cloud  0.6.3
llama-index-legacy                       0.9.48.post4
llama-index-llms-ollama                  0.4.1
llama-index-llms-openai                  0.3.2
llama-index-multi-modal-llms-ollama      0.4.0
llama-index-multi-modal-llms-openai      0.3.0
llama-index-postprocessor-rankllm-rerank 0.4.0
llama-index-program-openai               0.3.1
llama-index-question-gen-openai          0.3.0
llama-index-readers-file                 0.4.0
llama-index-readers-llama-parse          0.4.0
llama-index-vector-stores-qdrant         0.4.0
llama-parse                              0.5.15
ollama                                   0.4.2

Using Jupyterlab via Nvidia AI workbench and the Ollama server running on Win 10

### Version

0.12.2

### Steps to Reproduce

For simplicity, use the https://docs.llamaindex.ai/en/v0.10.19/examples/multi_modal/ollama_cookbook.html example:

from llama_index.multi_modal_llms.ollama import OllamaMultiModal
mm_model = OllamaMultiModal(model="llava:v1.6",base_url='http://host.docker.internal:11434')
from pathlib import Path
from llama_index.core import SimpleDirectoryReader
from PIL import Image
import matplotlib.pyplot as plt

input_image_path = Path("restaurant_images")
if not input_image_path.exists():
    Path.mkdir(input_image_path)

!wget "https://docs.google.com/uc?export=download&id=1GlqcNJhGGbwLKjJK1QJ_nyswCTQ2K2Fq" -O ./restaurant_images/fried_chicken.png

# load as image documents
image_documents = SimpleDirectoryReader("./restaurant_images").load_data()
# display image
imageUrl = "./restaurant_images/fried_chicken.png"
image = Image.open(imageUrl).convert("RGB")
plt.figure(figsize=(16, 5))
plt.imshow(image)

from pydantic import BaseModel

class Restaurant(BaseModel):
    """Data model for an restaurant."""

    restaurant: str
    food: str
    discount: str
    price: str
    rating: str
    review: str

from llama_index.core.program import MultiModalLLMCompletionProgram
from llama_index.core.output_parsers import PydanticOutputParser

prompt_template_str = """\
{query_str}

Return the answer as a Pydantic object. The Pydantic schema is given below:

"""
mm_program = MultiModalLLMCompletionProgram.from_defaults(
    output_parser=PydanticOutputParser(Restaurant),
    image_documents=image_documents,
    prompt_template_str=prompt_template_str,
    multi_modal_llm=mm_model,
    verbose=True,
)
response = mm_program(query_str="Can you summarize what is in the image?")
for res in response:
    print(res)

### Relevant Logs/Tracbacks

```shell
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[8], line 1
----> 1 response = mm_program(query_str="Can you summarize what is in the image?")
      2 for res in response:
      3     print(res)

File ~/.local/lib/python3.10/site-packages/llama_index/core/instrumentation/dispatcher.py:321, in Dispatcher.span.<locals>.wrapper(func, instance, args, kwargs)
    318             _logger.debug(f"Failed to reset active_span_id: {e}")
    320 try:
--> 321     result = func(*args, **kwargs)
    322     if isinstance(result, asyncio.Future):
    323         # If the result is a Future, wrap it
    324         new_future = asyncio.ensure_future(result)

File ~/.local/lib/python3.10/site-packages/llama_index/core/program/multi_modal_llm_program.py:104, in MultiModalLLMCompletionProgram.__call__(self, llm_kwargs, image_documents, *args, **kwargs)
    101 llm_kwargs = llm_kwargs or {}
    102 formatted_prompt = self._prompt.format(llm=self._multi_modal_llm, **kwargs)  # type: ignore
--> 104 response = self._multi_modal_llm.complete(
    105     formatted_prompt,
    106     image_documents=image_documents or self._image_documents,
    107     **llm_kwargs,
    108 )
    110 raw_output = response.text
    111 if self._verbose:

File ~/.local/lib/python3.10/site-packages/llama_index/core/instrumentation/dispatcher.py:321, in Dispatcher.span.<locals>.wrapper(func, instance, args, kwargs)
    318             _logger.debug(f"Failed to reset active_span_id: {e}")
    320 try:
--> 321     result = func(*args, **kwargs)
    322     if isinstance(result, asyncio.Future):
    323         # If the result is a Future, wrap it
    324         new_future = asyncio.ensure_future(result)

File ~/.local/lib/python3.10/site-packages/llama_index/core/llms/callbacks.py:431, in llm_completion_callback.<locals>.wrap.<locals>.wrapped_llm_predict(_self, *args, **kwargs)
    422 event_id = callback_manager.on_event_start(
    423     CBEventType.LLM,
    424     payload={
   (...)
    428     },
    429 )
    430 try:
--> 431     f_return_val = f(_self, *args, **kwargs)
    432 except BaseException as e:
    433     callback_manager.on_event_end(
    434         CBEventType.LLM,
    435         payload={EventPayload.EXCEPTION: e},
    436         event_id=event_id,
    437     )

File ~/.local/lib/python3.10/site-packages/llama_index/multi_modal_llms/ollama/base.py:176, in OllamaMultiModal.complete(self, prompt, image_documents, formatted, **kwargs)
    164 """Complete."""
    165 response = self._client.generate(
    166     model=self.model,
    167     prompt=prompt,
   (...)
    171     **kwargs,
    172 )
    173 return CompletionResponse(
    174     text=response["response"],
    175     raw=response,
--> 176     additional_kwargs=get_additional_kwargs(response, ("response",)),
    177 )

File ~/.local/lib/python3.10/site-packages/llama_index/multi_modal_llms/ollama/base.py:28, in get_additional_kwargs(response, exclude)
     25 def get_additional_kwargs(
     26     response: Dict[str, Any], exclude: Tuple[str, ...]
     27 ) -> Dict[str, Any]:
---> 28     return {k: v for k, v in response.items() if k not in exclude}

File ~/.local/lib/python3.10/site-packages/pydantic/main.py:856, in BaseModel.__getattr__(self, item)
    853     return super().__getattribute__(item)  # Raises AttributeError if appropriate
    854 else:
    855     # this is the current error
--> 856     raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')

AttributeError: 'GenerateResponse' object has no attribute 'items'
```

**Comment by logan-markewich** [comment]
Ollama had some breaking changes on v0.4 of the client. Try downgrading. I'll have to make a patch

**Comment by baxteran** [comment]
Hi @logan-markewich - I've got quite a bunch of dependencies to manage, so for clarity - are you asking me to downgrade:

llama-index-llms-ollama                  0.4.1
llama-index-multi-modal-llms-ollama      0.4.0
ollama                                   0.4.2

Do you have any recommendations for version?

Many thanks for you speedy reply and support!

Andy.

**Comment by baxteran** [comment]
Downgrading to ollama-0.3.3 works for now, thanks.

**Comment by KartikeyShaurya36** [comment]
> 0.3.3

Hi @baxteran  IT is not working after downgrading also

**Comment by ermannocavalli** [comment]
Hi @logan-markewich I want to use llama3.2-vision, which is compatible only with ollama > 0.4. Any idea when the patch will be released?

**Comment by VDChinhs** [comment]
@baxteran Why downgrade ollama-0.3.3, I think wrong configuration in package llama-index-multi-modal-llms-ollama

**Comment by baxteran** [comment]
> [@baxteran](https://github.com/baxteran) Why downgrade ollama-0.3.3, I think wrong configuration in package llama-index-multi-modal-llms-ollama

Oh interesting. In what way do you think this is misconfigured? 

Thanks.

**Comment by VDChinhs** [comment]
@baxteran I don't know but I tried ollama 0.3.3 it worked.
Tks.

**Comment by baxteran** [comment]
@logan-markewich similar to @ermannocavalli I would like to use the new multi-modal features. Appreciate you are very busy, but do you have a rough idea when a patch may be available? Thanks for all your help!

**Comment by baxteran** [comment]
@VDChinhs - Hi - I get you now. ollama-0.3.3 works fine for me too, it's the newer versions that caused issue. Thanks!

**Comment by JackWittmayer** [comment]
I'm not sure what exactly changed between ollama versions but the GenerateResponse object no longer behaves like a dict. I think the code fix is something like changing `response.items()` in [this line](https://github.com/run-llama/llama_index/blob/main/llama-index-integrations/multi_modal_llms/llama-index-multi-modal-llms-ollama/llama_index/multi_modal_llms/ollama/base.py#L28) to `response.model_dump().items()`. I made this override locally and the query engine works now.

I might make a pull request if I have time.

**Comment by JJLee1215** [comment]
@JackWittmayer  
Your solution is great! Now my code works well too. Thanks!"

**Comment by ajstair** [comment]
Confirming that this is still happening with Ollama 0.6.4. 

An override works, as mentioned above. @JackWittmayer did you ever make a PR? 

(I opted to just use the ollama library directly to generate a completion.)

**Comment by csttsn** [comment]
I'm having the same issue with LlamaIndex 0.12.41 and Ollama version 0.9.0. And I really hope that LlamaIndex developers can look into it and fix it in the foreseeable future.

Regarding the workaround mentioned in [this comment](https://github.com/run-llama/llama_index/issues/17105#issuecomment-2620053317), unfortunately, now changing `response.items()` to `response.model_dump().items()` is not enough; instead, I had to update the calls within the `complete `and `acomplete` methods to use `self._get_additional_kwargs` as follows:
```
class PatchedOllamaMultiModal(OllamaMultiModal):
    @staticmethod
    def _get_additional_kwargs(
        response: Dict[str, Any], exclude: Tuple[str, ...]
    ) -> Dict[str, Any]:
        """Get additional kwargs from response."""
        return {k: v for k, v in response.items() if k not in exclude}

    def complete(
        self,
        prompt: str,
        image_documents: Sequence[ImageNode],
        **kwargs: Any,
    ) -> CompletionResponse:
        from llama_index.core.multi_modal_llms.generic_utils import (
            image_documents_to_base64,
        )

        response_obj = self._client.generate(
            model=self.model,
            prompt=prompt,
            images=image_documents_to_base64(image_documents),
            stream=False,
            options=self._model_kwargs,
            **kwargs,
        )

        # The PATCH: convert the Pydantic model to a dict
        response_dict = response_obj.model_dump()

        return CompletionResponse(
            text=response_dict["response"],
            raw=response_dict,
            additional_kwargs=self._get_additional_kwargs(response_dict, ("response",)), 
        )

    async def acomplete(
        self,
        prompt: str,
        image_documents: Sequence[ImageNode],
        **kwargs: Any,
    ) -> CompletionResponse:
        from llama_index.core.multi_modal_llms.generic_utils import (
            image_documents_to_base64,
        )

        response_obj = await self._aclient.generate(
            model=self.model,
            prompt=prompt,
            images=image_documents_to_base64(image_documents),
            stream=False,
            options=self._model_kwargs,
            **kwargs,
        )

        # The PATCH: convert the Pydantic model to a dict
        response_dict = response_obj.model_dump()

        return CompletionResponse(
            text=response_dict["response"],
            raw=response_dict,
            additional_kwargs=self._get_additional_kwargs(response_dict, ("response",)),
        )
```

**Comment by anishesg** [comment]
PR up at https://github.com/run-llama/llama_index/pull/21543.
